### DISCUSSION TITLE
Support for sops decryption methods other than GPG?

### DISCUSSION THREAD

**Discussion by brokenjacobs** [fluxcd/flux2#277] [Q&A]
Is there a plan to support other decryption methods besides GPG? Specifically KMS solutions from the various cloud vendors.

**Comment by stefanprodan** [comment]
The gpg secret is not required, if you bind an IAM role to the default service account where you installed the toolkit then it should work, I mentioned the KMS solutions here https://toolkit.fluxcd.io/guides/mozilla-sops/#configure-secrets-decryption

**Reply by brokenjacobs** [reply]
I thought that support was missing from the merged sops code when I looked at the kustomize-controller source code.

**Reply by brokenjacobs** [reply]
I'm trying to use aws kms with an on premises cluster as well so I don't think I can bind the service account to a role like that, may have to use env vars.
